spec/models/generic_announcement_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe GenericAnnouncement, type: :model do
+  describe '.valid' do
+    before do
+      create_list(:system_announcement, 3)
```

Comment:
`before...create` might be better written as `let!(:identifier) { ... }`. See http://betterspecs.org @allenwq